src/parser/grammar.lalrpop

```diff
@@ -508,21 +508,10 @@ Interpolation = { "%{", "multstr %{" };
 
 StaticString: String = StringStart <s: ChunkLiteral?> StringEnd => s.unwrap_or_default();
 
-EnumTag: Ident = "`" <EnumTagContent> => <>;
-
-EnumTagContent: Ident = {
-    <Ident>,
-    <StaticString> => <>.into(),
-    // Allow tags of the form `Str, `Bool, etc. This is ad-hoc and should
-    // eventually be replaced with a proper lexing of enum tags, but this is not
-    // a trivial task. For now, to make builtin.typeof usable, we just
-    // special-case builtin types.
-    "Dyn" => "Dyn".into(),
-    "Num" => "Num".into(),
-    "Bool" => "Bool".into(),
-    "Str" => "Str".into(),
-    "Array" => "Array".into(),
-}
+EnumTag: Ident = {
+  "raw enum tag" => <>.into(),
+  "`\"" <s: ChunkLiteral?> "\"" => s.unwrap_or_default().into(),
```

Comment:
It's not very important, but I wonder if we can factor this out using the `StaticString` rule, which is more or less the same. The issue is that the start and end rule for the literal are different, but we can parametrize that with a macro rule in LARLPOP:

```rust
StaticString<Start, End> : String = Start <s: ChunkLiteral?> StrigEnd => s.unwrap_or_default();
```

Then replace all current occurrences of `StaticString` with `StaticString<StringStart, StringEnd>` and the one above with

```rust
StaticString<"`\"", "\""> => <>.into(),
```

Or use a different name `StaticStringDelim` and just write `StaticString = StaticStringDelim<StringStart, StringEnd>`. I'm not sure any of these are better than the current one, but once again, it's rather to let you know that those are also possibilities :slightly_smiling_face: 

It wouldn't change much the complexity of the current code, but could be a tad useful if we decide that multiline string delimiters for `import` also don't make sense, using e.g. `StaticStringDelim<"\"", "\"">` instead of `StaticString`. It's also more abstraction for something quite simple, though.

Reply:
Thank you! The duplication was bothering me, but it hadn't occurred to me that I could just factor it out a macro. Will definitely do this before I merge. :pray: